web/app/scripts/localization/date-picker-directive.js

```diff
@@ -0,0 +1,95 @@
+(function() {
+    'use strict';
+
+    /* ngInject */
+    function DatePickerController(DateLocalization, $log, $scope, $element) {
+        var ctl = this;
+
+        ctl.$onInit = initialize;
+
+        function initialize() {
+            // functions
+            ctl.updateDate = updateDate;
+
+            // data
+            ctl.dateConfig = DateLocalization.currentDateFormats();
+            ctl.internalCalendar = $.calendars.instance(
+                ctl.dateConfig.calendar, ctl.dateConfig.language
+            );
+
+            // $scope
+            // initialize with passed in datetime, then cancel watch so local changes don't cause a loop
+            $scope.$watch(function() {return ctl.datetime;}, function(date) {
+                if (typeof(date) === 'undefined') {
+                    return;
+                }
+                $($element).calendarsPicker(
+                    'setDate', ctl.internalCalendar.fromJSDate(date)
+                );
+            });
+        }
+
+        function updateDate(date) {
+            ctl.internalDateTime = ctl.internalCalendar.formatDate(ctl.calendarOptions.dateFormat, date);
```

Comment:
I'm not sure I understand how this field is being used; it looks like it gets updated by `ng-model` and `updateDate` but I don't see where the data ever gets read back out or used to calculate anything.